webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/mesh/NeuroglancerMeshHelper.scala

```diff
@@ -0,0 +1,183 @@
+    meshFormat: String,
+    lods: List[MeshLodInfo],
+    chunkScale: Array[Double] = Array(1.0, 1.0, 1.0) // Used for Neuroglancer Precomputed Meshes to account for vertex quantization
+)
+
+object WebknossosSegmentInfo {
+  implicit val jsonFormat: OFormat[WebknossosSegmentInfo] = Json.format[WebknossosSegmentInfo]
+
+  def fromMeshInfosAndMetadata(chunkInfos: List[List[MeshLodInfo]],
+                               encoding: String,
+                               chunkScale: Array[Double] = Array(1.0, 1.0, 1.0)): Option[WebknossosSegmentInfo] =
+    chunkInfos.headOption.flatMap { firstChunkInfo =>
+      tryo {
+        WebknossosSegmentInfo(
+          meshFormat = encoding,
+          lods = chunkInfos.transpose.map(mergeLod),
+          chunkScale = chunkScale
+        )
+      }
+    }
+
+  private def mergeLod(thisLodFromAllChunks: List[MeshLodInfo]): MeshLodInfo = {
+    val first = thisLodFromAllChunks.head
+    first.copy(chunks = thisLodFromAllChunks.flatMap(_.chunks))
+  }
+
+}
+trait NeuroglancerMeshHelper {
+
+  def computeGlobalPosition(segmentInfo: NeuroglancerSegmentManifest,
+                            lod: Int,
+                            lodScaleMultiplier: Double,
+                            currentChunk: Int): Vec3Float
+
+  def getLodTransform(segmentInfo: NeuroglancerSegmentManifest,
+                      lodScaleMultiplier: Double,
+                      transform: Array[Array[Double]],
+                      lod: Int): Array[Array[Double]]
+
+  protected def enrichSegmentInfo(segmentInfo: NeuroglancerSegmentManifest,
+                                  lodScaleMultiplier: Double,
+                                  transform: Array[Array[Double]],
+                                  neuroglancerOffsetStart: Long,
+                                  segmentId: Long): List[MeshLodInfo] = {
+    val bytesPerLod = segmentInfo.chunkByteSizes.map(_.sum)
+    val totalMeshSize = bytesPerLod.sum
+    val meshByteStartOffset = neuroglancerOffsetStart - totalMeshSize
+    val chunkByteOffsetsInLod = segmentInfo.chunkByteSizes.map(_.scanLeft(0L)(_ + _)) // builds cumulative sum
+
+    def getChunkByteOffset(lod: Int, currentChunk: Int): Long =
+      // get past the finer lods first, then take offset in selected lod
+      bytesPerLod.take(lod).sum + chunkByteOffsetsInLod(lod)(currentChunk)
+
+    def computeGlobalPositionAndOffset(lod: Int, currentChunk: Int): MeshChunk = {
+      val globalPosition = computeGlobalPosition(segmentInfo, lod, lodScaleMultiplier, currentChunk)
+
+      MeshChunk(
+        position = globalPosition, // This position is in Voxel Space
+        byteOffset = meshByteStartOffset + getChunkByteOffset(lod, currentChunk),
+        byteSize = segmentInfo.chunkByteSizes(lod)(currentChunk).toInt, // size must be int32 to fit in java array
+        unmappedSegmentId = Some(segmentId)
+      )
+    }
+
+    val lods: Seq[Int] = for (lod <- 0 until segmentInfo.numLods) yield lod
```

Comment:
```suggestion
    val lods: Seq[Int] = 0 until segmentInfo.numLods
```
I think that’s equivalent